### ISSUE TITLE
Streamlabs IOS app cannot connect RTMP

### ISSUE THREAD

**Issue by dbussert** [bluenviron/mediamtx#2352]
## Which version are you using?

v1.0.3

## Which operating system are you using?

- [X] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

Using the IOS StreamLabs app to send RTMP to the server, the connection will never connect. It initiates, hangs and times out.

The behavior is similar to this resolved issue with the StreamLabs Android app. https://github.com/bluenviron/mediamtx/issues/1244. I have confirmed the current Android app connects fine, the issue is only with the IOS app.

## Describe how to replicate the issue

1. Download StreamLabs app for Iphone
2. Create `Custom RTMP` stream to MediaMTX
3. Observe stream never connects

## Did you attach the server logs?

yes 

```
2023/09/15 14:12:10 INF [RTMP] [conn 192.168.1.134:65316] opened
2023/09/15 14:12:11 DEB [path asdf] created
2023/09/15 14:12:20 DEB [path asdf] destroyed (not in use)
2023/09/15 14:12:20 INF [RTMP] [conn 192.168.1.134:65316] closed (read tcp 10.5.1.75:1936->192.168.1.134:65316: i/o timeout)
```

## Did you attach a network dump?

no

**Comment by aler9** [comment]
Hello, first of all, the RTMP implementation was recently improved, please try the nightly release attached in this thread, it may fix this issue:
https://github.com/bluenviron/mediamtx/issues/2289#issuecomment-1705702518

If it doesn't work, attach a network dump.

**Comment by dbussert** [comment]
> Hello, first of all, the RTMP implementation was recently improved, please try the nightly release attached in this thread, it may fix this issue: [#2289 (comment)](https://github.com/bluenviron/mediamtx/issues/2289#issuecomment-1705702518)
> 
> If it doesn't work, attach a network dump.

The nightly build did fix camera streaming with the StreamLabs app, so that's positive. The screen sharing feature of StreamLabs still did not work however. I've attached a pcap of the screen-sharing connection attempt.
[streamlabs.zip](https://github.com/bluenviron/mediamtx/files/12661089/streamlabs.zip)

**Comment by aler9** [comment]
Thanks for reporting the screen sharing issue, this is fixed by #2611
